docs/examples/docs_mcp/docs_mcp_server.py

```
@@ -0,0 +1,464 @@
+    case_sensitive: bool = False,
+    context_lines: int = 1,
+    max_matches: int = 50,
+) -> GrepFilterResult:
+    """Filter documentation content by grep-like pattern. Returns structured result.
+
+    Use this after ibm_search_doc to narrow down specific lines (e.g. config keys,
+    error codes, API endpoints). Supports regex patterns.
+
+    Examples:
+        filter_grep(content, r"timeout|retry")               # timeout or retry
+        filter_grep(content, r"^\s*- ", max_matches=20)      # bullet points
```

Ruff (0.15.6)

[warning] Invalid escape sequence: `\s`

Add backslash to escape sequence

(W605)